src/App.js

```diff
@@ -10,29 +10,35 @@ class App extends Component {
     super();
     this.state = {
       movies:[],
-       error:null
+
+      error:null,
+      selected: null
+
     }
     
   }
 
   showMore = (id) => {
-    const selectedMovie = this.state.movies.filter(movie => movie.id === id)
-    this.setState({movies: selectedMovie})
-    // render() {
-    //   return (
-    //   <SingleView />
-    //   )
-    // }
+    const selectedMovie = this.state.movies.find(movie => movie.id === id)
+    this.setState({ selected: selectedMovie })
+  }
+
+  showAll = () => {
+    console.log('SHIT SHIT SHIT')
+    this.setState({ selected: null })
   }
 
   
 
   componentDidMount = () => {
-    console.log('App, componentDidMount')
+    // console.log('App, componentDidMount')
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
       .then((response) => response.json())
       .then((data) => { 
         this.setState({ movies: data.movies })
+
+        // console.log('the state',this.state)
+
       })
       .catch((error) => {
         this.setState({error: 'The server is experincing some issues please try  again later?'}) 
```

Comment:
Love that you're setting state for the error key in the catch block - now make sure you're throwing an Error on 400 responses to ensure it hits that!